cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -5039,6 +5054,154 @@ if(ARROW_S3)
+      -DENABLE_TESTING=OFF
+      -DENABLE_UNITY_BUILD=ON
+      -DWARNINGS_AS_ERRORS=OFF)
+  
+  STRING(REPLACE "." ";" OPENSSL_VERSION_ELEMENTS_LIST ${OPENSSL_VERSION}) 
+  list(GET OPENSSL_VERSION_ELEMENTS_LIST 0 OPENSSL_VERSION_MAJOR)
+  if(OPENSSL_VERSION_MAJOR EQUAL 1)
+    list(APPEND
+    AZURESDK_COMMON_CMAKE_ARGS
+    -DVCPKG_OVERLAY_PORTS=${CMAKE_CURRENT_BINARY_DIR}/azuresdk_ep-prefix/src/azuresdk_ep/vcpkg-custom-ports)
+  elseif(OPENSSL_VERSION_MAJOR EQUAL 3)
+  else()
+    message(FATAL_ERROR "Unknown OpenSSL version: ${OPENSSL_VERSION}")
+  endif()
+
+  file(MAKE_DIRECTORY ${AZURESDK_INCLUDE_DIR})
+  set(AZURE_CORE_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-core${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_IDENTITY_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-identity${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_STORAGE_BLOBS_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-storage-blobs${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_STORAGE_COMMON_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-storage-common${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  set(AZURE_STORAGE_FILES_DATALAKE_STATIC_LIBRARY
+      "${AZURESDK_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}azure-storage-files-datalake${CMAKE_STATIC_LIBRARY_SUFFIX}"
+  )
+  externalproject_add(azuresdk_ep
+                      ${EP_COMMON_OPTIONS}
+                      INSTALL_DIR ${AZURESDK_PREFIX}
+                      URL ${ARROW_AZURESDK_URL}
+                      URL_HASH "SHA256=${ARROW_AZURE_SDK_BUILD_SHA256_CHECKSUM}"
+                      CMAKE_ARGS ${AZURESDK_COMMON_CMAKE_ARGS}
+                      BUILD_BYPRODUCTS ${AZURE_CORE_STATIC_LIBRARY}
+                                       ${AZURE_IDENTITY_STATIC_LIBRARY}
+                                       ${AZURE_STORAGE_BLOBS_STATIC_LIBRARY}
+                                       ${AZURE_STORAGE_COMMON_STATIC_LIBRARY}
+                                       ${AZURE_STORAGE_FILES_DATALAKE_STATIC_LIBRARY}
+                      DEPENDS azure_sdk_dependencies)
+  add_library(Azure::azure-core STATIC IMPORTED)
+  set_target_properties(Azure::azure-core
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_CORE_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-core
+               PROPERTY INTERFACE_LINK_LIBRARIES CURL::libcurl OpenSSL::Crypto OpenSSL::SSL)
+  add_dependencies(Azure::azure-core azuresdk_ep)
+
+  add_library(Azure::azure-identity STATIC IMPORTED)
+  set_target_properties(Azure::azure-identity 
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_IDENTITY_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
```

`Azure::azure-core` provides it:

```suggestion
                        PROPERTIES IMPORTED_LOCATION ${AZURE_IDENTITY_STATIC_LIBRARY})
```

---

Again, I'm happy to make this change if you feel strongly, but from my perspective it makes sense to have an explicit dependency like this even if its redundant with another transitive dependency.